app/javascript/components/supplemental_application/actions.js

```diff
@@ -1,18 +1,35 @@
 import apiService from '~/apiService'
 import domainToApi from '~/components/mappers/domainToApi'
 import Alerts from '~/components/Alerts'
-import { isEmpty } from 'lodash'
+import { isEmpty, find, isEqual, every } from 'lodash'
 
-export const updateApplication = async (application) => {
-  const leasePromise = updateLease(
-    application['lease'], application['primaryApplicantContact'], application['id']
-  )
+export const updateApplication = async (application, prevApplication) => {
+  const promises = [updateLease(application['lease'], application['primaryApplicantContact'], application['id'])]
 
+  // Conctact lease promise with rental assistances
+  promises.concat(updateRentalAssistances(application, prevApplication))
   const applicationApi = domainToApi.buildApplicationShape(application)
-  const appPromise = apiService.submitApplication(applicationApi)
+  promises.push(apiService.submitApplication(applicationApi))
+  const responses = await Promise.all(promises)
 
-  const [appResponse, leaseResponse] = await Promise.all([appPromise, leasePromise])
-  return appResponse !== false && leaseResponse !== false
+  return every(responses, (promise) => promise !== false)
```

Comment:
what does this line do?

Reply:
it evaluates every promise to be !== false. Before we had only two promises coming from api requests, now we can have more as each rental assistance update is a separate request.